src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserGroupService.kt

```diff
@@ -61,9 +65,11 @@ class UserGroupService(
   }
 
   private suspend fun userSecurityCheck(userId: UUID) {
-    userRepository.findById(userId)?.let { u: User ->
-      maintainUserCheck.ensureUserLoggedInUserRelationship(u.name)
-    } ?: throw UsernameNotFoundException("User $userId not found")
+    if (!hasViewUserGroupsRole(authenticationFacade.getAuthentication().authorities)) {
```

Comment:
Think it would be safer to move this check out of `userSecurityCheck` function and use it to suppress the call to `userSecurityCheck` from `getParentGroups` and `getAllGroupsUsingChildGroupsInLieuOfParentGroup`

Reply:
I think we said we could do as Alan suggested OR given `userSecurityCheck` is only being used in this service for read only endpoints, rename `userSecurityCheck` -> `checkUserGroupsReadAccess` or similar

Reply:
moved checks closed to method